lib/std/mem.zig

```diff
@@ -688,6 +805,7 @@ test "len" {
     }
 }
 
+/// Deprecated: use std.mem.len() or std.mem.scanZ().len
```

Comment:
If you ban sentinels other than `0` then we shouldn't deprecate this.

Reply:
I'd argue that `spanZ` is the wrong name for a function that works with sentinels other than 0. I also don't think there is high demand for working with any other sentinel in practice. If there is in fact demand, my preferred solution would be adding a `scanSentinel()` function that takes the sentinel type and value as arguments.

Reply:
> I also don't think there is high demand for working with any other sentinel in practice

to the contrary, I work with other sentinel values regularly. Most commonly `null` to indicate the end of a list of pointers passed from C; but also other fields, e.g. when working with [`luaL_Reg`](https://www.lua.org/manual/5.4/manual.html#luaL_Reg) the other day:
> Any array of luaL_Reg must end with a sentinel entry in which both name and func are NULL.

Reply:
I think it would be worse than the existing `lenZ`/`spanZ`: there's no need to pass in the sentinel as it's already part of the type of the pointer.

Reply:
@daurnimator if the sentinel is part of the pointer, then `len()`/`span()` work fine.